### DISCUSSION TITLE
Using an anchor style button within a paragraph w/ auto layout

### DISCUSSION THREAD

**Discussion by ceyerkes** [twilio-labs/paste#1855] [General] [answered]
Hi! I was wondering how to handle using an anchor style button within a paragraph when autolayout is enabled:

![Screen Shot 2021-09-10 at 11 49 25 AM](https://user-images.githubusercontent.com/87141551/132896142-644cd1de-5a29-483f-9781-c72bebf84f72.png)

Currently, I just manually change the text style within the paragraph to blue/underlined, create some space, and paste in the anchor icon - just wondering if there's a better way to do this. Thanks!

**Comment by nkrantz** [comment, marked as answer]
Hi @ceyerkes! What is the Link-style Button doing here? Does it redirect the user to a new URL on click?

**Reply by ceyerkes** [reply]
hey @nkrantz! yes, it links out to https://www.campaignregistry.com/ on click.

**Reply by nkrantz** [reply]
Is there a reason you're using a Button here? Generally, if clicking navigates the user to a new URL, we recommend using an [Anchor component](https://paste.twilio.design/components/anchor). This helps screen readers and other assistive technologies better communicate what the outcome of a user's action will be. With an Anchor, you can automatically add in the icon (that I think you're referencing) by adding the `showExternal` prop.

If there's a separate reason you need to use a Button here instead of an Anchor, please let me know!

**Reply by ceyerkes** [reply]
Hi Nora - sorry for the confusion, that is an anchor component - I didn't include the anchor icon because I was trying to communicate that I can't add one within the body of the paragraph - here's another example:

![Screen Shot 2021-09-13 at 10 27 31 AM](https://user-images.githubusercontent.com/87141551/133121634-8a58e031-be60-4ea0-bb8f-c03e428fdbb4.png)

Let me know if that makes more sense!

**Reply by ceyerkes** [reply]
Might be a bit easier to explain "in person" if you have a couple minutes to discuss, just let me know!

**Reply by nkrantz** [reply]
Ah ok! If you'd like to walk through it over zoom, feel free to sign up for Office Hours (we still have 2 open slots for this Thursday). Otherwise, could you provide a code sandbox link? That would help us understand what you're trying to achieve here.

We typically like to "work in the open" and limit our support to places that it can be reused (documented here on GH Discussions, in a recorded office hours session). Happy to look at a sandbox though!

**Reply by nkrantz** [reply]
OH signup link: https://docs.google.com/spreadsheets/d/10vD91zdwr9hjgDOV3HSGnoC0LgvYw-thmn3X_kvW0xk/edit#gid=664218049

**Reply by ceyerkes** [reply]
@nkrantz sounds good, I'll sign up for office hours!

**Reply by ceyerkes** [reply]
I found an example on the Paste website, which might help further explain the issue we're running into:

![Screen Shot 2021-09-13 at 4 35 09 PM](https://user-images.githubusercontent.com/87141551/133165191-d5253338-6a88-45e1-9bcd-25e05ca75dd3.png)

I just wanted to confirm that this is a limitation of Figma, and see if you have a workaround on your end - inserting an anchor link within body copy when auto-layout is being used. You can't drag in an anchor component inside of paragraph copy. I can still sign up for office hours and explain more, but that's the gist. Does that make more sense? Thanks @nkrantz!

**Reply by nkrantz** [reply]
@ceyerkes, apologies! I thought that you were having difficulty implementing the Anchor component in code. I've touched base with our design team, and Katie made a quick video to explain a couple of work-arounds. I hope this helps! https://www.loom.com/share/34902c03da354ce69fa8ecf0a4f394ee

**Reply by ceyerkes** [reply]
Thanks so much @nkrantz and @katieporter! That video is super helpful. I'll share with my team!

**Reply by ceyerkes** [reply]
Hey Nora! Reopening this, because I thought of another solution that I wanted to run by the Paste team. This is still a frustrating Figma issue (though the solution by @katieporter is a great workaround) and I'm looking into using font glyphs so that my components that use autolayout don't completely break when the copy changes. Is this something that's been looked into yet? I've read a few articles on the Figma website, and it seems like the most straightforward option, assuming we can find icons that match what we need. The ones that I use that are typically embedded in a paragraph are the anchor and info icons.

https://help.figma.com/hc/en-us/articles/360040449513-Add-icons-to-text-layers-with-icon-fonts

https://forum.figma.com/t/multi-line-text-icon-in-line-auto-layout/13153

**Reply by ceyerkes** [reply]
Also - let me know if I should open a new discussion.

**Reply by shleewhite** [reply]
Thank you for bringing this option up! Using the external-link-alt glyph from Font Awesome, or even a glyph like ↗ could definitely work, but you would need to be clear when you hand off the design to engineers that they should use [the Paste External Anchor component](https://paste.twilio.design/components/anchor#external-anchor) and not the glyph.

We’ve also tossed around the idea of creating a custom icon font for Paste iconography to solve for this use case, but for now, we’ve determined that the upfront cost and ongoing maintenance probably aren’t worth the payoff. But if this is something that you think you and other PDs would get a huge benefit from, definitely let us know!

**Reply by ceyerkes** [reply]
Ok sounds good! I brought this up in the DSYS Committee meeting earlier this week, and it sounds like it's something that other PDs would benefit from. I'm happy to throw it out there on the UX Channel too, to see what the interest is there is as well. It's really just two icons that I'm running into daily that I would love an icon font for (anchor and info), in case that reduces maintenance concerns?

Zhen also shared [this resource](https://fontello.com/) for generating icon fonts, which I'll try out as well. Thanks!
